### ISSUE TITLE
Remove BPM limits when entering with keyboard or +/- buttons

### ISSUE THREAD

**Issue by ggttttggggtt** [Kr0oked/Metronome#11]
Though rare, there is music slower than 40 and faster than 208 BPM.

**Comment by Kr0oked** [comment]
For the moment I will keep the current range. Unfortunately it gets harder to select a specific value on the slider when the range is bigger. So I think it is more important to easily select common tempos then allow very slow or very fast tempos.

**Comment by ggttttggggtt** [comment]
On my phone it lets me select the BPM text box here, and the keyboard comes up, but input is blocked.

**Comment by luzidd** [comment]
I also think it would be good to have arbitrary bpm limits.
Maybe have the values for MIN & MAX configurable in the Settings.

**Comment by Kr0oked** [comment]
> I also think it would be good to have arbitrary bpm limits. Maybe have the values for MIN & MAX configurable in the Settings.

I like the approach with the settings, maybe I will implement it that way

**Comment by Rstment25** [comment]
Please fix this, let us set the limits manually...

**Comment by Kr0oked** [comment]
I decided against implementing an arbitrary tempo selection. From a technical perspective and also from the UI perspective I want to have a consistent model. The slider, input field and the buttons should allow the exact same values.

But I acknowledge that the tempo range from 40-208 bmp is too small for some scenarios. These values were adopted from the range of classical [mechanical metronomes](https://upload.wikimedia.org/wikipedia/commons/5/5f/Metronome_Nikko.jpg).

In the latest release [1.4.5](https://github.com/Kr0oked/Metronome/releases/tag/v1.4.5) I extended the range to 30-252 bpm. Now I adopted the tempo range from a modern commercial grade metronome, the [Korg KDM-3](https://cdn.korg.com/in/products/upload/adb130190c2a3653b9c0f65fcccc5dee_pc.png).

**Comment by ksdrummer** [comment]
Hi, thank you for letting us know.

As a professional musician I know it's not rare at all to use metronome below this limit, I do it every day and many of my friends, my students also do it. In my opinion, it just makes you app limited... what makes this app different from the other ones if it doesn't offer anything more intersting to the users? 

But I respect you choice, and wish you the best, I love the look and feel of this metronome.

For those who need a range below 30 BPM, I strongly recommend Chubby Click from f-droid. It's been my main metronome for a while now.

**Comment by Rstment25** [comment]
> I decided against implementing an arbitrary tempo selection. From a technical perspective and also from the UI perspective I want to have a consistent model. The slider, input field and the buttons should allow the exact same values.
> 
> But I acknowledge that the tempo range from 40-208 bmp is too small for some scenarios. These values were adopted from the range of classical [mechanical metronomes](https://upload.wikimedia.org/wikipedia/commons/5/5f/Metronome_Nikko.jpg).
> 
> In the latest release [1.4.5](https://github.com/Kr0oked/Metronome/releases/tag/v1.4.5) I extended the range to 30-252 bpm. Now I adopted the tempo range from a modern commercial grade metronome, the [Korg KDM-3](https://cdn.korg.com/in/products/upload/adb130190c2a3653b9c0f65fcccc5dee_pc.png).

Ok but why?

It makes your app unsightly or something along those lines? 

Why replicate it 1to1? Why not make something better...

**Comment by Kr0oked** [comment]
Yes, one of my concerns is that the "simple" operability suffers.
And I dont think limitations are a bad thing. Actually the app is better if the values are limited to the ones needed for a metronom. The app is not a toy but a tool for a specific use case.

If I allow a tempo of 900 bpm you sure have more freedom, but you also have a noise generator and not a metronom.

**Comment by droid192** [comment]
please let the user go below 30, just why!?
https://github.com/Kr0oked/Metronome/blob/da48852b1d59b355a08543f4e5b69e153109c39e/app/src/main/java/com/bobek/metronome/data/Tempo.kt#L32

**Comment by ksdrummer** [comment]
Hey I'm just here because I received an update on this thread. I don't use this app anymore (because of the limitations), and I didn't answer before because I felt Kr00ked answer was slightly disrespectful. But anyways...

You don't want to change the limitations of your app, and it's totally fine. You're making your metronome the way you want. Good for you !

However, as a professional musician, I need my metronome to go below 10BPM. Other metronomes like the excellent chubby click do the job so no problem, however no need to answer our suggestion to improve your app the way you did.

No hard feelings though, I wish you the best with this promising metronome (it looks good at least).
